.github/workflows/release.yml

```diff
@@ -240,45 +212,47 @@ jobs:
           name: spin
           path: checksums-${{ env.RELEASE_VERSION }}.txt
 
-      - name: upload checksums to Github release
-        if: startsWith(github.ref, 'refs/tags/v')
-        uses: svenstaro/upload-release-action@v2
-        with:
-          repo_token: ${{ secrets.GITHUB_TOKEN }}
-          file: checksums-${{ env.RELEASE_VERSION }}.txt
-          tag: ${{ github.ref }}
-          prerelease: ${{ needs.build-and-sign.outputs.prerelease == 'true' }}
-
-  update-canary:
-    name: update canary release
+  create-gh-release:
+    name: create GitHub release
     runs-on: ubuntu-latest
     needs: checksums
-    if: github.ref == 'refs/heads/main'
+    env:
+      GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
     steps:
-      - name: Download release assets
+      - uses: actions/checkout@v3
+
+      - name: download release assets
         uses: actions/download-artifact@v3
         with:
           name: spin
+          path: _dist
 
-      - name: Delete canary tag
-        uses: dev-drprasad/delete-tag-and-release@v0.2.1
-        env:
-          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
-        with:
-          tag_name: canary
-          delete_release: true
+      - name: check if pre-release
+        shell: bash
+        run: |
+          if [[ ! "${{ github.ref_name }}" =~ ^v[0-9]+.[0-9]+.[0-9]+$ ]]
+          then
+            echo "PRERELEASE=--prerelease" >> "$GITHUB_ENV"
+          fi
 
-      - name: Recreate canary tag and release
-        uses: ncipollo/release-action@v1.13.0
-        with:
-          tag: canary
-          allowUpdates: true
-          prerelease: true
-          artifacts: "checksums-canary.txt,spin-canary*"
-          commit: ${{ github.sha }}
-          body: |
+      - name: create GitHub release (canary)
+        if: github.ref == 'refs/heads/main'
+        run: |
+          gh release delete canary --cleanup-tag
```

Comment:
+100 for using gh cli instead of third party actions for creating canary release